File: 2-Add-Two-Numbers/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int sum = l1.val + l2.val;
        int val = sum % 10;
        int carry = sum / 10;
        ListNode result = new ListNode(val);
        ListNode p = result;
        ListNode nextNode;
        l1 = l1.next;
        l2 = l2.next;
        while (l1 != null && l2 != null) {
            sum = l1.val + l2.val + carry;
            val = sum % 10;
            carry = sum / 10;
            nextNode = new ListNode(val);
            p.next = nextNode;
            p = nextNode;
            l1 = l1.next;
            l2 = l2.next;
        }
        ListNode n = (l1 != null ? l1 : l2);
        while (n != null) {
            sum = n.val + carry;
            val = sum % 10;
            carry = sum / 10;
            nextNode = new ListNode(val);
            p.next = nextNode;
            p = nextNode;
            n = n.next;
            
        }
        if (carry != 0) {
            p.next = new ListNode(carry);
        }
        return result;
    }
}

File: 2-Add-Two-Numbers/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        head = ListNode(0)
        output = head
        carry = 0
        while True:
            if l1 != None:
                carry = l1.val + carry
                l1 = l1.next
            if l2 != None:
                carry = l2.val + carry
                l2 = l2.next
            output.val = carry % 10
            carry = carry / 10
            if l1 != None or l2 != None or carry:
                output.next = ListNode(0)
                output = output.next
            else:
                break
        return head
            
            
        